src/misc/setup.cpp

```diff
@@ -983,18 +983,26 @@ const Section_line &Config::GetOverwrittenAutoexecSection() const
 
 bool Config::ParseConfigFile(const std::string &type, const std::string &configfilename)
 {
+	std::error_code ec;
+	const std_fs::path cfg_path = configfilename;
+	const auto conanical_path = std_fs::canonical(cfg_path, ec);
+	if (ec)
+		return false;
+	
+	if (contains(configFilesCanonical, conanical_path)) {
+		LOG_INFO("CONFIG: Skipping duplicate config file '%s'", 
+			configfilename.c_str());
+		return true;
+	}
 	// static bool first_configfile = true;
-	ifstream in(configfilename);
-	if (!in)
+	ifstream in(conanical_path);
+	if (!in) 
```

Comment:
(very minor: extra space after `(!in)`.  Assuming the last commit goes through CI, can you squash the last two together?  (unless you think there's value in having the `string` version available too)

Reply:
Nah, the path version is better. More efficient.

Thanks for noticing that.

I'll do a fixup once CI has passed. (It should pass, previous "failures" have been a canceled run, and MacOS failures when downloading a dependency)

Reply:
> Nah, the path version is better. More efficient.
> 
> Thanks for noticing that.
> 
> I'll do a fixup once CI has passed. (It should pass, previous "failures" have been a canceled run, and MacOS failures when downloading a dependency)

Oooh look, as I speak, a MacOS failure with dependency issue....